hubble/src/eth/indexer.rs

```diff
@@ -64,18 +69,17 @@ impl Config {
         let provider = RaceClient::new(
             self.urls
                 .into_iter()
-                .map(|url| Provider::<Http>::try_from(url.as_str()).unwrap())
+                .map(|url| ProviderBuilder::new().on_http(url))
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/unionlabs/union/pull/2693#issuecomment-2293655644>#ef6b49</a><span> - 08/16/2024, 03:05 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The new provider initialization using 'ProviderBuilder::new().on_http(url)' does not handle potential errors that might occur during the initialization process. This could lead to runtime panics if the URL is invalid or if there are network issues. A similar issue was also found in hubble/src/eth/fork.rs (line 51).
</div>
<div id="fix">
<b>Fix:</b> Add proper error handling for the provider initialization to ensure that any issues during the initialization process are gracefully handled.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -72,7 +72,9 @@
 -                .map(|url| ProviderBuilder::new().on_http(url))
 +                .map(|url| {
 +                    ProviderBuilder::new().on_http(url).map_err(|e| {
 +                        error!("Failed to initialize provider: {:?}", e);
 +                        e
 +                    })
 +                })
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged